torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -731,6 +745,56 @@ TensorView* clamp(TensorView* in, Val* min_val, Val* max_val) {
   return clamp(in->as<Val>(), min_val, max_val)->as<TensorView>();
 }
 
+// sum_to operator
+
+inline bool shouldReduceSumTo(const int shape, const Val* root_id_extent) {
+  return shape == 1 && !root_id_extent->isOneInt();
+}
+
+inline bool shouldReduceSumTo(const Val* shape, const Val* root_id_extent) {
+  return shape->isOneInt() && !root_id_extent->isOneInt();
+}
+
+// Name of sum_to is different from NV fuser naming,
+// this is to align with the operator name
+template <typename T>
+TensorView* sum_to_impl(TensorView* v1, const std::vector<T>& shape) {
+  TensorView* v2 = v1;
+
+  const auto& v1_root = TensorDomain::noReductions(v1->getRootDomain());
+
+  const int64_t leading_dims = v1_root.size() - shape.size();
```

Comment:
Can you add an assert `v1_root.size() >= shape.size()`?

Reply:
Added. Thanks!